php-frontend/src/main/java/org/sonar/php/utils/collections/ListUtils.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.utils.collections;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ListUtils {
+
+  private ListUtils() {
+  }
+
+  @SafeVarargs
+  public static <T> List<T> concat(List<? extends T>... lists) {
+    return Arrays.stream(lists)
+      .flatMap(List::stream)
+      .collect(Collectors.toList());
+  }
```

Comment:
Streams always add a little bit of performance overhead.
I think I wouldn't have used them in a utility method which may be used in a lot of places.
Maybe that would be considered premature optimization.
Same thing for `SetUtils`.

Reply:
You're absolutely right. It's better to think about the effects on performance at an early stage.